core/domain/exp_domain.py

```diff
@@ -2499,3 +2499,470 @@ def add_contribution_by_user(self, contributor_id):
+                            if (STATE_PROPERTY_CONTENT not in
+                                    self.changed_translations[state_name] and
+                                    STATE_PROPERTY_RECORDED_VOICEOVERS not in
+                                    self.changed_properties[state_name]):
+                                change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_INTERACTION_ID):
+                    if state_name in self.changed_properties:
+                        if (old_exp_states.interaction.id ==
+                                current_exp_states.interaction.id):
+                            if (STATE_PROPERTY_INTERACTION_CUST_ARGS
+                                    not in
+                                    self.changed_properties[state_name] and
+                                    STATE_PROPERTY_INTERACTION_ANSWER_GROUPS
+                                    not in
+                                    self.changed_properties[state_name] and
+                                    STATE_PROPERTY_INTERACTION_SOLUTION
+                                    not in
+                                    self.changed_properties[state_name]):
+                                change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                # Customization args differ for every interaction, so in
+                # case of different interactions merging is simply not
+                # possible, but in case of same interaction, the values in
+                # the customization_args are often lists so if someone
+                # changes even one item of that list then determining which
+                # item is changed is not feasible, so suppose there is long
+                # list of values in item selection interaction and one user
+                # deletes one value and another one edits another value,
+                # so after deletion the indices of all the values will be
+                # changed and it will not be possible to compare and know
+                # that which value is changed by second user.
+                # So we will not be handling the merge on the basis of
+                # individual fields.
+                elif (change.property_name ==
+                      STATE_PROPERTY_INTERACTION_CUST_ARGS):
+                    if state_name in self.changed_properties:
+                        if (old_exp_states.interaction.id ==
+                                current_exp_states.interaction.id):
+                            if (all(property not in
+                                    self.changed_properties[state_name]
+                                    for property in
+                                    (self
+                                     .PROPERTIES_CONFLICTING_CUST_ARGS_CHANGES
```

Comment:
**Improvement:** Find a consistent way to handle conflict properties (I see multiple duplicate structures of if-else case in al the PROPERTIES_CONFLICTING_*)